sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+
+describe(
+  describeProcessorRootTest(QueueNames.CancelApplicationAssessment),
+  () => {
+    let app: INestApplication;
+    let processor: CancelApplicationAssessmentProcessor;
+    let zbClientMock: ZBClient;
+    let appDataSource: DataSource;
+    let studentAssessmentRepo: Repository<StudentAssessment>;
+    let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+    let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, zbClient } =
+        await createTestingAppModule();
+      app = nestApplication;
+      zbClientMock = zbClient;
+      appDataSource = dataSource;
+      studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+      disbursementOverawardRepo = dataSource.getRepository(
+        DisbursementOveraward,
+      );
+      disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+      // Processor under test.
+      processor = app.get(CancelApplicationAssessmentProcessor);
+    });
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it("Should cancel the assessment pending disbursements and rollback overawards when the cancelled application has overawards and also one sent and one pending disbursements.", async () => {
+      // Arrange
+      const workflowInstanceId = faker.random.number({
+        min: 1000000000,
+        max: 9999999999,
+      });
+      // Application and disbursements.
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        null,
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          createSecondDisbursement: true,
+        },
+      );
+      // Adjust assessment.
+      application.applicationStatus = ApplicationStatus.Cancelled;
+      const studentAssessment = application.currentAssessment;
+      studentAssessment.assessmentWorkflowId = workflowInstanceId.toString();
+      await studentAssessmentRepo.save(application.currentAssessment);
+      // Adjust disbursements.
+      const [firstDisbursement, secondDisbursement] =
+        studentAssessment.disbursementSchedules;
+      firstDisbursement.disbursementScheduleStatus =
+        DisbursementScheduleStatus.Sent;
+      await disbursementScheduleRepo.save(firstDisbursement);
+      // Add a student overaward.
+      const overaward = createFakeDisbursementOveraward({
+        student: application.student,
+        studentAssessment,
+      });
+      await disbursementOverawardRepo.save(overaward);
+      // Queued job.
+      const job = createMock<Job<CancelAssessmentQueueInDTO>>({
+        data: { assessmentId: studentAssessment.id },
+      });
+
+      // Act
+      const result = await processor.cancelAssessment(job);
+
+      // Assert
+      expect(zbClientMock.cancelProcessInstance).toBeCalledWith(
+        workflowInstanceId.toString(),
+      );
+      expect(result.summary).toContain(
```

Comment:
We can expect the whole object as we do in other e2e tests like this if the order of summary values are guaranteed.

![image](https://user-images.githubusercontent.com/54600590/231592785-2d633598-4758-4cd8-b558-8e316cc31a06.png)

Reply:
I do not have a hard feeling IMO, we are matching what we want for the test in the PR.

Reply:
I made the change.
I was unwilling to validate the complete output for the queue consumers because, different from the API E2E where the return data is the most essential part, I considered the processing and DB as the most important and the output log as secondary information.